src/Parser/EnhancementShaman/CombatLogParser.js

```diff
@@ -0,0 +1,336 @@
+import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
+
+import SmolderingHeart from './Modules/Legendaries/SmolderingHeart';
+
+import './Modules/Main/main.css';
+
+import CPM_ABILITIES, { SPELL_CATEGORY } from './CPM_ABILITIES';
+
+function formatThousands(number) {
+  return (Math.round(number || 0) + '').replace(/(\d)(?=(\d{3})+(?!\d))/g, "$1,");
+}
+
+function formatNumber(number) {
+  if (number > 1000000) {
+    return `${(number / 1000000).toFixed(2)}m`;
+  }
+  if (number > 10000) {
+    return `${Math.round(number / 1000)}k`;
+  }
+  return formatThousands(number);
+}
+
+function getIssueImportance(value, regular, major, higherIsWorse = false) {
+  if (higherIsWorse ? value > major : value < major) {
+    return ISSUE_IMPORTANCE.MAJOR;
+  }
+  if (higherIsWorse ? value > regular : value < regular) {
+    return ISSUE_IMPORTANCE.REGULAR;
+  }
+  return ISSUE_IMPORTANCE.MINOR;
+}
+function formatPercentage(percentage) {
+  return (Math.round((percentage || 0) * 10000) / 100).toFixed(2);
+}
+
+class CombatLogParser extends MainCombatLogParser {
+  static specModules = {
+    // Features
+    alwaysBeCasting: AlwaysBeCasting,
+    cooldownTracker: CooldownTracker,
+    procTracker: ProcTracker,
+
+    // Legendaries:
+  };
+
+  generateResults() {
+    const results = super.generateResults();
+
+    //first row of talents
+    const hasWindSong = this.selectedCombatant.hasTalent(SPELLS.WINDSONG_TALENT.id);
+    const hasHotHand = this.selectedCombatant.hasTalent(SPELLS.HOT_HAND_TALENT.id);
+    const hasLandslide = this.selectedCombatant.hasTalent(SPELLS.LANDSLIDE_TALENT.id);
+    //4th row of talents
+    const hasHailstorm = this.selectedCombatant.hasTalent(SPELLS.HAILSTORM_TALENT.id);
+    //5th row of talents
+    const hasOvercharge = this.selectedCombatant.hasTalent(SPELLS.HAILSTORM_TALENT.id);
+    //6th row of talents
+    const hasCrashingStorm = this.selectedCombatant.hasTalent(SPELLS.CRASHING_STORM_TALENT.id);
+    const hasFuryOfAir = this.selectedCombatant.hasTalent(SPELLS.FURY_OF_AIR_TALENT.id);
+    const hasSundering = this.selectedCombatant.hasTalent(SPELLS.SUNDERING_TALENT.id);
+    //last row of talents
+    const hasAscendance = this.selectedCombatant.hasTalent(SPELLS.ASCENDANCE_TALENT_ENHANCEMENT.id);
+    const hasEarthenSpike = this.selectedCombatant.hasTalent(SPELLS.EARTHEN_SPIKE_TALENT.id);
+    const hasBoulderfist = this.selectedCombatant.hasTalent(SPELLS.BOULDERFIST_TALENT.id);
+
+    const abilityTracker = this.modules.abilityTracker;
+    const getAbility = spellId => abilityTracker.getAbility(spellId);
+
+    const flametongue = getAbility(SPELLS.FLAMETONGUE.id);
+    const frostbrand = getAbility(SPELLS.FROSTBRAND.id);
+    const stormBringer = getAbility(SPELLS.STORMBRINGER.id);
+
+    const fightDuration = this.fightDuration;
+
+	  //uptimes
+	  const flametongueUptime = this.selectedCombatant.getBuffUptime(SPELLS.FLAMETONGUE_BUFF.id) / this.fightDuration;
```

Comment:
Seems to have added a tab here, does your IDE have settings to insert 2 spaces when you hit the tab key? (WebStorm does this by default) Don't spend too much time on this, I'll implement Prettier that should fix this as soon as it has proper JSX support.